api/controllers/EventController.js

```diff
@@ -89,16 +96,46 @@ var EventController = {
     },
 
     getEvents: function(req, res) {
+        var sortOption = req.swagger.params.sortBy.value;
         var page = req.swagger.params.page.value;
         var pageSize = req.swagger.params.pageSize.value;
 
-        Event.find({ 'hasFinished': false }, function (e, events){
-            if (e) return res.status(500).json(e.message);
-
-            res.json({events: events});
-        })
-        .skip((page-1)*pageSize)
-        .limit(pageSize)
+        var lng = req.swagger.params.lng.value;
+        var lat = req.swagger.params.lat.value;
+        var radius = req.swagger.params.radius.value;
+        
+        if(lat && lng){
+            Event.find(
+                { location :
+                    { $near :
+                        {
+                            $geometry : {
+                                type : "Point" ,
+                                coordinates : [lng, lat] },
+                            $maxDistance : radius
+                        }
+                    }
+                }, function (e, events){
+                    if (e) return res.status(500).json(e.message);
+
+                    res.json({events: events});
+            })
+            .sort(sortOption)
```

Comment:
might not be a good idea passing `sortOption` in directly, for example if the input from api user is "hahahah" it should return an error

Reply:
for the sortBy parameter, should we use a Integer instead?
Like 0 for distance? 1 for order by createdAt?

Reply:
there are pros and cons, a set of integer might be convenient when if-checking them but we'll have to maintain an enum for both frontend and backend, also it's not quite readable when you see `sortBy=0`.